### PR TITLE
Disk Access changes

### PR DESCRIPTION
**Pull request by haagha**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by grizzlytheodore** [comment]
needs changelog.md

**Comment by msJinLei** [comment]
@haagha  Have you submitted a cmdlet review https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues?
If yes, could you provide the link?

**Comment by grizzlytheodore** [comment]
@msJinLei  https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/639 
here is the design review. 
I added 'Do not merge' label, we need to demo to our stakeholders and make sure they are happy with what we implemented.

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 1]
```diff
@@ -921,3 +921,161 @@ function Test-DiskEncryptionSet
...
+        New-AzResourceGroup -Name $rgname -Location $loc -Force;
+
+        #Testing disk access
+        $diskAccess = New-AzDiskAccess -ResourceGroupName $rgname -Name "diskaccessname" -location $loc
+        $diskconfig = New-AzDiskConfig -Location $loc -SkuName 'Standard_LRS' -OsType 'Windows' `
+                                        -UploadSizeInBytes 35183298347520 -CreateOption 'Upload' -DiskAccessId $diskAccess.Id;
+        New-AzDisk -ResourceGroupName $rgname -DiskName $diskname0 -Disk $diskconfig;
+        $disk = Get-AzDisk -ResourceGroupName $rgname -DiskName $diskname0;
+        
+        Assert-AreEqual $diskAccess.Id $disk.DiskAccessId;
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1018]: $disk.DiskAccessID I am getting null for this value. not sure why it's passing. 

Assert-NotNull $disk.DiskAccessID may help before this

Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1018]: @haagha

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 [hunk 2]
```diff
@@ -921,3 +921,161 @@ function Test-DiskEncryptionSet
...
+    {
+        # Common
+        $loc = Get-ComputeVMLocation;
+        New-AzResourceGroup -Name $rgname -Location $loc -Force;
+        $subId = Get-SubscriptionIdFromResourceGroup $rgname;
+        $mocksourcevault = '/subscriptions/' + $subId + '/resourceGroups/' + $rgname + '/providers/Microsoft.KeyVault/vaults/TestVault123';
+        $mockkey = 'https://myvault.vault-int.azure-int.net/keys/mockkey/00000000000000000000000000000000';
+        $mocksecret = 'https://myvault.vault-int.azure-int.net/secrets/mocksecret/00000000000000000000000000000000';
+        $access = 'Read';
+
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1052]: this part above isnt used so I think they can be taken out @Sandido

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1018]: nevermind. worked second time

File: src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1 (hunk 2 above)
Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1134]: @haagha correction

Comment (grizzlytheodore) [reply src/Compute/Compute.Test/ScenarioTests/DiskRPTests.ps1:1129]: i removed it

File: src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs [hunk 3]
```diff
@@ -165,6 +165,16 @@ public partial class NewAzureRmDiskConfigCommand : Microsoft.Azure.Commands.Reso
...
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string DiskAccessId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string NetworkAccessPolicy { get; set; }
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs:176]: add completer

File: src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs [hunk 4]
```diff
@@ -353,8 +353,8 @@ private void Run()
                 CreationData = vCreationData,
                 EncryptionSettingsCollection = vEncryptionSettingsCollection,
                 Encryption = vEncryption,
-                NetworkAccessPolicy = NetworkAccessPolicy,
-                DiskAccessId = DiskAccessId
+                NetworkAccessPolicy = this.IsParameterBound(c => c.NetworkAccessPolicy) ? this.NetworkAccessPolicy : null,
+                DiskAccessId = this.IsParameterBound(c => c.DiskAccessId) ? this.DiskAccessId : null
```
[the remark this answers is not in the document]
Comment (haagha) [reply src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs:358]: checked

File: src/Compute/Compute/help/New-AzDiskConfig.md [hunk 5]
```diff
@@ -231,7 +231,8 @@ Accept wildcard characters: False
 ```
 
 ### -DiskAccessId
-Id of DiskAccess.
+Gets or sets ARM id of the DiskAccess resource for using private endpoints on.
```
[the remark this answers is not in the document]
Comment (haagha) [reply src/Compute/Compute/help/New-AzDiskConfig.md]: edited

File: src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs (hunk 3 above)
Comment (haagha) [reply src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs:177]: added

**Comment by Sandido** [comment]
@msJinLei , we have addressed the review comments. Please review this.

File: src/Compute/Compute/ChangeLog.md [hunk 6]
```diff
@@ -17,8 +17,12 @@
     ## YYYY.MM.DD - Version X.Y.Z (Previous Release)
     * Overview of change #1
         - Additional information about change #1
--->
```
Comment (msJinLei) [inline src/Compute/Compute/ChangeLog.md:20]: You should add back -->

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by msJinLei** [comment]
@haagha 
- Please resolve the conflicts and fix the error in ChangeLog.md

**Comment by grizzlytheodore** [comment]
@msJinLei thank you for catching that. your comment has been addressed

File: src/Compute/Compute/ChangeLog.md [hunk 7]
```diff
@@ -18,11 +18,15 @@
...
         - Additional information about change #1
 
+-->
 ## Upcoming Release
 * Added the '-EncryptionType' optional parameter to New-AzVmDiskEncryptionSetConfig.
 * New cmdlets for new resource type: DiskAccess 'Get-AzDiskAccess', 'New-AzDiskAccess' , 'Get-AzDiskAccess'
 * Added optional parameters '-DiskAccessId' and '-NetworkAccessPolicy' to New-AzSnapshotConfig
 * Added optional parameters '-DiskAccessId' and '-NetworkAccessPolicy' to New-AzDiskConfig
+* Added 'AssignedHost' field to Get-AzVM and Get-AzVmss's instance views. The field shows the resource id of the virtual machine instance
+* Added `SupportAutomaticPlacement` to New-AzHostGroup
```
Comment (Sandido) [inline src/Compute/Compute/ChangeLog.md:28]: "Added the optional parameter 'SupportAutomaticPlacement' to New-AzHostGroup"

**Comment by grizzlytheodore** [comment]
@msJinLei please review and merge

File: src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs [hunk 8]
```diff
@@ -129,6 +129,16 @@ public partial class NewAzureRmSnapshotConfigCommand : Microsoft.Azure.Commands.
...
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string DiskAccessId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true)]
+        public string NetworkAccessPolicy { get; set; }
```
Comment (msJinLei) [inline src/Compute/Compute/Generated/Snapshot/Config/NewAzureRmSnapshotConfigCommand.cs:140]: Why NetworkAccessPolicy has no argument completer?

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/help/New-AzDiskEncryptionSetConfig.md [hunk 9]
```diff
@@ -48,6 +48,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -EncryptionType
```
Comment (msJinLei) [inline src/Compute/Compute/help/New-AzDiskEncryptionSetConfig.md:51]: It is very strange that argument completer is not generated here. Can you try to regenerate it here?

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by grizzlytheodore** [comment]
@msJinLei  your comments are addressed. We appreciate it. Please review and merge!

**Review by msJinLei — APPROVED** [review]

**Review by msJinLei — APPROVED** [review]

**Review by msJinLei — APPROVED** [review]

**Comment by msJinLei** [comment]
@haagha please waiting for the sign of from @Sandido

**Review by Sandido — APPROVED** [review]

**Comment by Sandido** [comment]
@msJinLei , I approved the PR.

**Pull request merged**